app/controllers/projects_controller.rb

```diff
@@ -18,18 +18,22 @@ def index
 
   # GET /projects/1
   def show
-    @project = Project.includes(:project_status, members: [:projects_users]).find(params[:id])
+    @project = Project.includes(:project_status, :categories, members: [:projects_users]).find(params[:id])
     @serializer_options[:include] = [:members]
     @serializer_options[:params] = { project_id: params[:id] }
     render json: ProjectSerializer.new(@project, @serializer_options).serialized_json
   end
 
   # POST /projects
   def create
+    @categories = Category.find(params[:project][:categories])
+
     @project = Project.new(project_params)
+
     @project.projects_users.build(admin: true, user_id: auth_user.id)
 
     if @project.save
+      @project.categories << @categories
```

Comment:
Fa una transazione sola così?

Reply:
Non è possibile costruire, ne creare a priori un record sulla tabella di join in una HABTM:[https://guides.rubyonrails.org/association_basics.html#methods-added-by-has-and-belongs-to-many](url)
anche se provi a crearlo ti blocca perchè l'entità principale, cioè Project (la Category invece è già lì) deve essere salvata prima della modifica nella tabella di join.

Tutto questo per dire che non esiste per quanto io sappia modo di accorpare in un'unica transazione una modifica di questo tipo(già di per se sconsigliata) -> [https://stackoverflow.com/questions/47126307/rails-5-add-record-to-join-table-without-model-has-and-belongs-to-many](url)

P.S. se mentre leggi pensi "non è possibile, allora a cosa serve?, sono davvero così dementi?",sappi che non sei l'unico [http://blog.flatironschool.com/why-you-dont-need-has-and-belongs-to-many/](url)

Reply:
30L solo con questa argomentazione

Reply:
❤️